tests/dragonfly/conftest.py

```diff
@@ -229,42 +230,89 @@ def memcached_connection(df_server: DflyInstance):
+    dfly_path = df_factory.dfly_path
+    tls_server_key_file_name = "df-key.pem"
+    tls_server_cert_file_name = "df-cert.pem"
     tls_server_key = dfly_path + tls_server_key_file_name
     tls_server_req = dfly_path + "df-req.pem"
-    step2 = rf'openssl req -newkey rsa:4096 -nodes -keyout {tls_server_key} -out {tls_server_req} -subj "/C=GR/ST=SKG/L=Thessaloniki/O=KK/OU=Comp/CN=Gr/emailAddress=does_not_exist@gmail.com"'
-    subprocess.run(step2, shell=True)
+    step1 = rf'openssl req -newkey rsa:4096 -nodes -keyout {tls_server_key} -out {tls_server_req} -subj "/C=GR/ST=SKG/L=Thessaloniki/O=KK/OU=Comp/CN=Gr/emailAddress=does_not_exist@gmail.com"'
+    subprocess.run(step1, shell=True)
 
-    # Step 3
     # Use CA's private key to sign dragonfly's CSR and get back the signed certificate
+    ca_key = gen_ca_cert["ca_key"]
+    ca_cert = gen_ca_cert["ca_cert"]
     tls_server_cert = dfly_path + tls_server_cert_file_name
-    step3 = fr'openssl x509 -req -in {tls_server_req} -days 1 -CA {ca_cert} -CAkey {ca_key} -CAcreateserial -out {tls_server_cert}'
-    subprocess.run(step3, shell=True)
-    return tls_server_key_file_name, tls_server_cert_file_name
+    step2 = fr'openssl x509 -req -in {tls_server_req} -days 1 -CA {ca_cert} -CAkey {ca_key} -CAcreateserial -out {tls_server_cert}'
+    subprocess.run(step2, shell=True)
+
+    return {"server_key": tls_server_key, "server_cert": tls_server_cert}
+
+
+@pytest.fixture(scope="session")
+def gen_client_cert(df_factory: DflyInstanceFactory, gen_ca_cert):
+    # Generate client certificates
+    tls_client_key_file_name = "client-key.pem"
+    dfly_path = df_factory.dfly_path
+    tls_client_key = dfly_path + tls_client_key_file_name
+
+    tls_client_req = dfly_path + "client-req.pem"
+    step1 = rf'openssl req -newkey rsa:4096 -nodes -keyout {tls_client_key} -out {tls_client_req} -subj "/C=It/ST=RM/L=Rome/O=KK/OU=Comp/CN=Gr/emailAddress=random@gmail.com"'
+    subprocess.run(step1, shell=True)
+
+    ca_key = gen_ca_cert["ca_key"]
+    ca_cert = gen_ca_cert["ca_cert"]
+    tls_client_cert_file_name = "client-cert.pem"
+    tls_client_cert = dfly_path + tls_client_cert_file_name
+    step2 = fr'openssl x509 -req -in {tls_client_req} -days 1 -CA {ca_cert} -CAkey {ca_key} -CAcreateserial -out {tls_client_cert}'
+    subprocess.run(step2, shell=True)
+
+    return {"client_key": tls_client_key, "client_cert": tls_client_cert}
 
 
 @pytest.fixture(scope="session")
-def with_tls_args(df_factory: DflyInstanceFactory, gen_tls_cert):
-    tls_server_key_file_name, tls_server_cert_file_name = gen_tls_cert
+def with_tls_server_args(gen_server_cert):
     args = {"tls": "",
-            "tls_key_file": df_factory.dfly_path + tls_server_key_file_name,
-            "tls_cert_file": df_factory.dfly_path + tls_server_cert_file_name,
-            "no_tls_on_admin_port": "true"}
+            "tls_key_file": gen_server_cert["server_key"],
```

Comment:
as discussed, this fixture is redundant as this args can be returned from gen_client_cert fixture